modules/vba_documents/app/controllers/vba_documents/v2/uploads_controller.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+require 'zip'
+
+require_dependency 'vba_documents/application_controller'
+require_dependency 'vba_documents/upload_error'
+require_dependency 'vba_documents/payload_manager'
+require_dependency 'vba_documents/upload_validator'
+require_dependency 'vba_documents/multipart_parser'
+require 'common/exceptions'
+
+module VBADocuments
+  module V2
+    class UploadsController < ApplicationController
+      include VBADocuments::UploadValidations
+      skip_before_action(:authenticate)
+
+      def submit
```

Comment:
Why are we streaming a file to our servers and then restreaming it to Amazon instead of uploading from user's browser directly to Amazon?

Reply:
That's a great question.  See this: